C++/C++.sublime-syntax

```diff
@@ -26,2340 +28,616 @@ first_line_match: |-
-      push:
-        - meta_content_scope: meta.preprocessor.macro.c++
-        - include: scope:source.c#preprocessor-line-continuation
-        - include: scope:source.c#preprocessor-line-ending
-        - include: scope:source.c#preprocessor-comments
-        - match: '({{identifier}})(?=\()'
-          scope: entity.name.function.preprocessor.c++
-          set:
-            - match: '\('
-              scope: punctuation.section.group.begin.c++
-              set: preprocessor-macro-params
-        - match: '{{identifier}}'
-          scope: entity.name.constant.preprocessor.c++
-          set: preprocessor-macro-definition
-
-  preprocessor-macro-params:
-    - meta_scope: meta.preprocessor.macro.parameters.c++ meta.group.c++
-    - match: '{{identifier}}'
-      scope: variable.parameter.c++
-    - match: \)
-      scope: punctuation.section.group.end.c++
-      set: preprocessor-macro-definition
-    - match: ','
-      scope: punctuation.separator.c++
-      push:
-        - match: '{{identifier}}'
-          scope: variable.parameter.c++
-          pop: true
-        - include: scope:source.c#preprocessor-line-continuation
-        - include: scope:source.c#preprocessor-comments
-        - match: '\.\.\.'
-          scope: keyword.operator.variadic.c++
-        - match: '(?=\))'
-          pop: true
-        - match: (/\*).*(\*/)
-          scope: comment.block.c++
-          captures:
-            1: punctuation.definition.comment.c++
-            2: punctuation.definition.comment.c++
-        - match: '\S+'
-          scope: invalid.illegal.unexpected-character.c++
-    - include: scope:source.c#preprocessor-line-continuation
-    - include: scope:source.c#preprocessor-comments
-    - match: '\.\.\.'
-      scope: keyword.operator.variadic.c++
-    - match: (/\*).*(\*/)
-      scope: comment.block.c++
-      captures:
-        1: punctuation.definition.comment.c++
-        2: punctuation.definition.comment.c++
-    - match: $\n
-      scope: invalid.illegal.unexpected-end-of-line.c++
-
-  preprocessor-macro-definition:
-    - meta_content_scope: meta.preprocessor.macro.c++
-    - include: scope:source.c#preprocessor-line-continuation
-    - include: scope:source.c#preprocessor-line-ending
-    - include: scope:source.c#preprocessor-comments
-    # Don't define blocks in define statements
-    - match: '\{'
-      scope: punctuation.section.block.begin.c++
-    - match: '\}'
-      scope: punctuation.section.block.end.c++
-    # Captures the namespace macro idiom
-    - match: '\b(namespace)\s+(?={{path_lookahead}}\s*\{)'
-      scope: meta.namespace.c++
+        1: keyword.control.exception.c++
+        2: meta.group.c++ punctuation.section.group.begin.c++
+      push: function-params
+    - match: \b(try|catch|finally|throw)\b
+      scope: keyword.control.exception.c++
+    - match: \bco_(await|yield|return)\b
+      scope: keyword.control.c++
+    - match: \b(requires)\s*(\()\b
```

Comment:
A `\(` is already a word boundary. Thus trailing `\b` is redundant or even may cause trouble.

```suggestion
    - match: \b(requires)\s*(\()
```